tasks/monitoring.yml

```diff
@@ -11,16 +11,21 @@
       dest: /tmp/add-monitoring-agent-repo.sh
       mode: 0755
 
-  - name: Add repo
+  - name: Add repo and install agent
     command:
       chdir: /tmp
-      cmd: bash add-monitoring-agent-repo.sh
+      cmd: "bash add-monitoring-agent-repo.sh --also-install --version={{ version }}"
+    environment:
+      REPO_CODENAME: "{{ ansible_distribution_release }}"
+    register: result
+    changed_when: "'No changes made.' not in result.stdout_lines"
     notify: restart monitoring agent
 
   when: package_state == "present"
```

Comment:
Nit: can you move the "when" to the top of the task? i.e.
```
- when: package_state == "present"
  block:
  - ...
```

As the block gets longer I think it's harder to tell when it's going to run if the when is at the bottom of the block